Automattic/wp-calypso
# `<SiteThumbnail>`: onload handling doesn't fire once component has unmounted

#### Proposed Changes

A warning in React said the `onload` handler was still being called after the component had unmounted.

* Remove `ref` in favor of image props
* Fixes the mShots on Safari

#### Testing Instructions



* Go to `/sites` in Safari browser
* Change between row and tile views
* Observe the mShots appear consistently
* Open the console, observe no warning or errors appear related to react memory leaks

#### Pre-merge Checklist






- Related to https://github.com/Automattic/wp-calypso/issues/67202
